services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -21,30 +21,39 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     val versionValue = version.map(_.name).getOrElse(config.defaultFileVersion)
 
     httpClient
-      .put(
+      .put[DummyResponseBody, UpdateFileRequest](
         fileUrl,
-        payload = Json.stringify(
-          Json.toJson(UpdateFileRequest(versionValue, projectFile.content, versionValue))
-        ),
+        payload = UpdateFileRequest(versionValue, projectFile.content, versionValue),
         headers = config.token
       )
       .flatMap {
         case Response(HttpStatusCodes.OK, _, _) =>
           Future.successful(Right("Success update file"))
         case Response(HttpStatusCodes.BadRequest, _, _) =>
           httpClient
-            .post(
+            .post[DummyResponseBody, UpdateFileRequest](
               fileUrl,
-              payload = Json.stringify(
-                Json.toJson(UpdateFileRequest(config.defaultFileVersion, projectFile.content, "Init commit"))
-              ),
+              payload = UpdateFileRequest(config.defaultFileVersion, projectFile.content, "Init commit"),
               headers = config.token
             )
             .map {
-              case Response(HttpStatusCodes.OK, _, _) => Right("Create new file")
-              case Response(_, body, _)               => Left(VersioningException(body))
+              case Response(HttpStatusCodes.OK, _, _)        => Right("Create new file")
+              case Response(_, FailureResponseBody(body), _) => Left(VersioningException(body))
+              case _ =>
+                Left(
+                  VersioningException(
+                    "status is successfull, but not described "
+                  )
+                )
             }
-        case Response(_, body, _) => Future.successful(Left(VersioningException(body)))
+        case Response(_, body, _) =>
+          body match {
+            case SuccessResponseBody(bodyInfo) =>
+              Future.successful(
+                Left(VersioningException("status codes processing may not correctly implemented: " + bodyInfo))
+              )
+            case FailureResponseBody(error) => Future.successful(Left(VersioningException(error)))
+          }
```

Comment:
Remove this nested pattern matching as you did in other places.

Reply:
done